prime-router/docs/standard-operating-procedures/adding-conformance-profile.md

```diff
@@ -0,0 +1,143 @@
+output can be transformed into a variety of formats dependent on the requirements of the receiver. This design  
+intentionally decouples the input from the output so that the formats of the two can be different.
+
+### Editing existing vs Creating New
+
+Integrating a new conformance profile involves editing existing data structures and mappings or creating new ones.
+Modifying an existing structures and mappings is preferred, but that option depends on the differences present in the
+new conformance profile. Here are a few example scenarios:
+
+- **The new profile only deprecates fields**
+    - No changes are needed to structures or mappings.
+- **The new profile adds new fields**
+    - The existing custom structure can be edited by adding the new fields. A new structure is not needed.
+- **The new profile changes the datatype of existing fields**
+    - If the datatypes are backwards compatible then the existing custom structure can be edited. Otherwise new
+      structures and mappings are needed.
+
+## Data Structures
+
+### HL7
+
+ReportStream uses the [HAPI HL7v2](https://github.com/hapifhir/hapi-hl7v2) library to parse HL7v2 messages. That
+library  
+includes structures from v2.1 to v2.8.1.
+
+#### Steps to edit structure:
+
+1. Find the existing custom structure to be edited. Current custom structure is
+   located `prime-router/src/main/java/fhirengine/translation/hl7/structures/radxmars251`
+2. Starting with the file located in the `/message` directory and named with the HL7 message type (ex ORU_R01,
+   ORM_O01,  
+   ADT_A01, etc.) proceed through the structure to find the class that needs to be altered.
+    - For example if there is a change to PID.7 you would open the `PATIENT_RESULT` class then `PATIENT` class
+      then `PID` class where the edit can be made.
+
+#### Steps to add structure:
+
+1. Add package to `prime-router/src/main/java/fhirengine/translation/hl7/structures/` to house structure
+2. Add a package named "message". Within that, add a java class named with the HL7 message type (ex ORU_R01, ORM_O01,  
+   ADT_A01, etc.). This package and class are required.
+3. Add a package named "group". Within that, add java classes for the HL7 groups present in the conformance profile.
+4. Add a package named "segment". Within that, add java classes for the HL7 segments present in the conformance profile.
+5. Add a package named "datatype". Within that, add java classes for the HL7 datatypes present in the conformance  
+   profile.
+
+- Review the java classes in the corresponding packages for other data structures to see examples of how those classes  
+  should be arranged.
+
+#### Choosing a Message Model
+
+Once the new structure is created the ReportStream code needs to be updated to use said structure. Steps to take are:
+
+1. `HL7Reader.getMessageProfile()` returns a MessageProfile object including two properties: typeID and profileID.
```

Comment:
Maybe a note that this might be subject to change (i.e. adding it to the topic/receiver instead), since we know we're not going to get the right values in MSH 21